parser/validator.bal

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
```

Comment:
If this file is added to avoid circular dependancies, shall we keep an issue to refactor `fhir-validator` to be independent from parser.  
Otherwise we will have duplicated code in here and in validator module

Reply:
https://github.com/wso2-enterprise/open-healthcare/issues/1669